things/policies/redis/policies.go

```diff
@@ -79,12 +79,12 @@ func (pc pcache) Remove(ctx context.Context, policy policies.Policy) error {
 
 // kv is used to create a key-value pair for caching.
 // If thingID is not empty, it will be appended to the value.
-func kv(p policies.Policy, thingID string) (string, string) {
-	if thingID != "" {
-		return p.Subject + separator + p.Object, strings.Join(p.Actions, separator) + separator + thingID
+func kv(p policies.CachedPolicy) (string, string) {
+	if p.ThingID != "" {
+		return p.Policy.Subject + separator + p.Policy.Object, strings.Join(p.Policy.Actions, separator) + separator + p.ThingID
```

Comment:
This looks a little more self-documenting:
```go
// kv is used to create a key-value pair for caching.
// If thingID is not empty, it will be appended to the value.
func kv(p policies.CachedPolicy) (string, string) {
	key := p.ThingKey + separator + p.ChannelID
	val := strings.Join(p.Policy.Actions, separator)
	if p.ThingID != "" {
		val += separator + p.ThingID
	}
	return key, val
}
```